src/service/viewport-impl.js

```diff
@@ -510,6 +502,33 @@ export class Viewport {
   }
 
   /**
+   * @param {{paddingTop: number, duration: (number|undefined), curve: (string|undefined)}} event
+   * @private
+   */
+  updateOnViewportEvent_(event) {
+    this.binding_.updateViewerViewport(this.viewer_);
+    const paddingTop = event.paddingTop;
+    if (paddingTop != this.paddingTop_) {
+      const lastPaddingTop = this.paddingTop_;
+      this.paddingTop_ = paddingTop;
+      this.binding_.updatePaddingTop(this.paddingTop_, /* adjustScroll */true,
+          lastPaddingTop);
+      this.fixedLayer_.updatePaddingTop(this.paddingTop_);
+
+      if (event.duration > 0) {
+        // Add transit effect on position fixed element
+        const tr = numeric(lastPaddingTop - this.paddingTop_, 0);
+        Animation.animate(this.win_.document.documentElement, time => {
+          const p = tr(time);
+          this.fixedLayer_.transformMutate(`translateY(${p}px)`);
+        }, event.duration, event.curve).thenAlways(() => {
+          this.fixedLayer_.transformMutate(null);
```

Comment:
The name would be kind of ugly then. We'd instead better call it `resetTransform`. But they way I looked at this - this is a glorified `setStyle(element, 'transform', X)` method, except it happens over an array of elements. So I wanted to give it a similar signature. FWIW, @muxin originally had two methods: `transform` and `resetTransform` :)